File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/holdings_auth.kt
// To parse the JSON, install Klaxon and do:
//
//   val holdingsAuth = HoldingsAuth.fromJson(jsonString)

package quicktype

import com.beust.klaxon.*

private val klaxon = Klaxon()

data class HoldingsAuth (
    @Json(name = "\$ref")
    val ref: String,

    @Json(name = "\$schema")
    val schema: String,

    val definitions: Definitions
) {
    public fun toJson() = klaxon.toJsonString(this)

    companion object {
        public fun fromJson(json: String) = klaxon.parse<HoldingsAuth>(json)
    }
}

data class Definitions (
    @Json(name = "Data")
    val data: Data,

    @Json(name = "HoldingsAuth")
    val holdingsAuth: HoldingsAuthClass
)

data class Data (
    val additionalProperties: Boolean,
    val properties: DataProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class DataProperties (
    @Json(name = "request_id")
    val requestID: RequestID
)

data class RequestID (
    val type: String
)

data class HoldingsAuthClass (
    val additionalProperties: Boolean,
    val properties: HoldingsAuthProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class HoldingsAuthProperties (
    val data: DataClass,
    val status: RequestID
)

data class DataClass (
    @Json(name = "\$ref")
    val ref: String
)

File: JsonSchemaAndStructs/KiteConnect-Rust-Structs/holdings_auth.rs
// Example code that deserializes and serializes the model.
// extern crate serde;
// #[macro_use]
// extern crate serde_derive;
// extern crate serde_json;
//
// use generated_module::[object Object];
//
// fn main() {
//     let json = r#"{"answer": 42}"#;
//     let model: [object Object] = serde_json::from_str(&json).unwrap();
// }

extern crate serde_derive;

#[derive(Serialize, Deserialize)]
pub struct HoldingsAuth {
    #[serde(rename = "$ref")]
    holdings_auth_ref: String,

    #[serde(rename = "$schema")]
    schema: String,

    #[serde(rename = "definitions")]
    definitions: Definitions,
}

#[derive(Serialize, Deserialize)]
pub struct Definitions {
    #[serde(rename = "Data")]
    data: Data,

    #[serde(rename = "HoldingsAuth")]
    holdings_auth: HoldingsAuthClass,
}

#[derive(Serialize, Deserialize)]
pub struct Data {
    #[serde(rename = "additionalProperties")]
    additional_properties: bool,

    #[serde(rename = "properties")]
    properties: DataProperties,

    #[serde(rename = "required")]
    required: Vec<String>,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "type")]
    data_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct DataProperties {
    #[serde(rename = "request_id")]
    request_id: RequestId,
}

#[derive(Serialize, Deserialize)]
pub struct RequestId {
    #[serde(rename = "type")]
    request_id_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct HoldingsAuthClass {
    #[serde(rename = "additionalProperties")]
    additional_properties: bool,

    #[serde(rename = "properties")]
    properties: HoldingsAuthProperties,

    #[serde(rename = "required")]
    required: Vec<String>,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "type")]
    holdings_auth_class_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct HoldingsAuthProperties {
    #[serde(rename = "data")]
    data: DataClass,

    #[serde(rename = "status")]
    status: RequestId,
}

#[derive(Serialize, Deserialize)]
pub struct DataClass {
    #[serde(rename = "$ref")]
    data_class_ref: String,
}
